### PR TITLE
Implementation of gpfs daemon network to the ansible toolkit.

### PR DESCRIPTION
**Pull request by Christop1964**
- Implementation of gpfs daemon network to the ansible toolkit.
  - Fix some typos inside the ansible yml files.
  - Fix problem with yum handler file.
  - Fix warning message with unnecessary brackets.

### CODE REVIEW

File: roles/core/cluster/tasks/cluster.yml [hunk 1]
```diff
@@ -43,6 +43,20 @@
...
+     groups: scale_daemon_network_nodes
+  when: hostvars[item]['scale_daemon_network'] is defined
+  with_items: "{{ ansible_play_hosts }}"
+
+- assert:
+    that: ansible_play_hosts|length == groups['scale_daemon_nodes']|length
+    msg: >-
+         Please specify the daemon network for all nodes in the cluster.
+  when: groups['scale_daemon_nodes'] is defined
+  run_once: true
```
Comment (acch) [inline roles/core/cluster/tasks/cluster.yml:58]: Does this mean that `scale_daemon_network` becomes a mandatory variable, in all cases?

File: roles/core/cluster/tasks/check.yml [hunk 2]
```diff
@@ -33,3 +33,20 @@
...
+      lineinfile:
+        path: /etc/ssh/sshd_config
+        state: present
+        regexp: '#ListenAddress 0.'
+        line: 'ListenAddress {{ ansible_fqdn }}'
+
+    - name: Restart ssh daemon
+      systemd:
+        name: sshd
+        state: reloaded
```
Comment (acch) [inline roles/core/cluster/tasks/check.yml:49]: This should be a handler, you might consider moving it to `roles/core/cluster/handlers/main.yml` and making sure that it *only* runs if 'sshd_config' got updated. We surely don't want to restart the SSH daemon on every run of the playbook...

https://docs.ansible.com/ansible/devel/user_guide/playbooks_handlers.html

File: roles/callhome/node/tasks/install.yml [hunk 3]
```diff
@@ -55,9 +55,12 @@
...
 
 - debug:
-        msg: "{{ scale_install_all_packages }}"
+    msg: "{{ scale_install_all_packages }}"
 
-- meta: flush_handlers
+- name: yum-clean-metadata
+  command: yum clean metadata
+  args:
+   warn: false
```
Comment (acch) [inline roles/callhome/node/tasks/install.yml:63]: What's the reason for this change? It seems to be unrelated to the other changes and unrelated to the daemon network config... Also, disabling warnings should only be used in very rare cases.

**Review by acch — CHANGES_REQUESTED** [review]
Hi Christoph, thanks for your contribution!

I don't fully understand the scenario that's addressed by your changes, can you provide a description and / or a sample for the documentation?

Thanks again!

File: roles/core/cluster/tasks/check.yml (hunk 2 above)
Comment (acch) [reply roles/core/cluster/tasks/check.yml:49]: One more remark: we already have code for adjusting SSH client & server configuration in `roles/core/precheck/tasks/prepare.yml`, and we already have a handler `reload-sshd`... you might want to move your code to that file and re-use the handler:

```
...
  - name: ...
    notify: reload-sshd
...
```

File: roles/core/cluster/tasks/cluster.yml (hunk 1 above)
Comment (Christop1964) [reply roles/core/cluster/tasks/cluster.yml:58]: Hi Achim,
sorry for my late response but I had a system crash on my laptop. 
Sure we have to update the documentation with that new feature. I will discuss this with Rajan.  

The scale_daemon_network is a optional variable that will be set at the inventory file on each cluster host
If a daemon network should be setup the customer has to specify the name of the network alias for the daemon network ip.

File: roles/core/cluster/tasks/check.yml (hunk 2 above)
Comment (Christop1964) [reply roles/core/cluster/tasks/check.yml:49]: Okay I will verify if this works using the existing handler.

File: roles/core/cluster/tasks/cluster.yml (hunk 1 above)
Comment (acch) [reply roles/core/cluster/tasks/cluster.yml:58]: Thanks for this explanation.

I understand that `scale_daemon_network` is optional. However, in this code block you're testing (asserting) that all hosts in the current play have this var defined... and that makes the var a mandatory one... or am I missing something?

File: roles/callhome/node/tasks/install.yml (hunk 3 above)
Comment (Christop1964) [reply roles/callhome/node/tasks/install.yml:63]: We got warnings at the screen, that the meta-handler isn't working because of a include statement .

[WARNING]: flush_handlers task does not support when conditional

As we moved it to here the warning was gone.
If you have a better solution let me know.

**Comment by acch** [comment]
Thanks for the additional details @Christop1964.

I'm afraid I still don't fully understand the scenario that this addresses. Currently we're using the `scale_daemon_nodename` for each node when creating the cluster. It defaults to `ansible_hostname`, but could be overridden (on a per-host basis) if users want to use a different hostname / network for GPFS daemon communication. What's the benefit of introducing the additional `scale_daemon_network` variable?

**Comment by rajan-mis** [comment]
> Thanks for the additional details @Christop1964.
> 
> I'm afraid I still don't fully understand the scenario that this addresses. Currently we're using the `scale_daemon_nodename` for each node when creating the cluster. It defaults to `ansible_hostname`, but could be overridden (on a per-host basis) if users want to use a different hostname / network for GPFS daemon communication. What's the benefit of introducing the additional `scale_daemon_network` variable?

@acch Thank you.  i think you are correct , user can override this flag based one their admin n/w or daemon n/w.

# hosts:
[cluster01]
host-vm1 scale_cluster_quorum=true   scale_cluster_manager=true scale_cluster_gui=false scale_daemon_nodename = <scale_daemon_nw>

@Christop1964 If this will work then we may not need some block of code speciifc to scale_daemon_nw . Can you please check.  Thanks

File: roles/callhome/node/tasks/install.yml (hunk 3 above)
Comment (acch) [reply roles/callhome/node/tasks/install.yml:63]: That's a good catch - thanks for spotting it!

I'd suggest to open a separate issue for it so we can discuss in more detail. This seems to be a general problem with the structure of the callhome role (conditional import of the file `install.yml`).

I don't think that replacing the handler with a plain command is appropriate to address this structural issue.

Comment (acch) [reply roles/callhome/node/tasks/install.yml:63]: I've opened #251 to document the issue.

**Comment by rajan-mis** [comment]
We fixed git issue with new PR , so closing this.
#259
